src/macros.rs

```diff
@@ -714,55 +726,83 @@ macro_rules! clap_app {
+                $crate::clap_app!{ @app ($crate::App::new(stringify!($name))) $($tail)* }
             ))
             $($tt)*
         }
     };
-// Yaml like function calls - used for setting various meta directly against the app
+    // Yaml like function calls - used for setting various meta directly against the app
     (@app ($builder:expr) ($ident:ident: $($v:expr),*) $($tt:tt)*) => {
-// clap_app!{ @app ($builder.$ident($($v),*)) $($tt)* }
         $crate::clap_app!{ @app
             ($builder.$ident($($v),*))
             $($tt)*
         }
     };
-
-// Add members to group and continue argument handling with the parent builder
+    // Add members to group and continue argument handling with the parent builder
     (@group ($builder:expr, $group:expr)) => { $builder.group($group) };
     // Treat the group builder as an argument to set its attributes
     (@group ($builder:expr, $group:expr) (@attributes $($attr:tt)*) $($tt:tt)*) => {
         $crate::clap_app!{ @group ($builder, $crate::clap_app!{ @arg ($group) (-) $($attr)* }) $($tt)* }
     };
+    (@group ($builder:expr, $group:expr) (@arg ($name:expr): $($tail:tt)*) $($tt:tt)*) => {
+        $crate::clap_app!{ @group
+            ($crate::clap_app!{ @app ($builder) (@arg ($name): $($tail)*) },
+             $group.arg($name))
+            $($tt)*
+        }
+    };
+    (@group ($builder:expr, $group:expr) (@arg $name:literal: $($tail:tt)*) $($tt:tt)*) => {
+        $crate::clap_app!{ @group
+            ($crate::clap_app!{ @app ($builder) (@arg $name: $($tail)*) },
+             $group.arg(stringify!($name).trim_matches('"')))
+            $($tt)*
+        }
+    };
     (@group ($builder:expr, $group:expr) (@arg $name:ident: $($tail:tt)*) $($tt:tt)*) => {
         $crate::clap_app!{ @group
             ($crate::clap_app!{ @app ($builder) (@arg $name: $($tail)*) },
              $group.arg(stringify!($name)))
             $($tt)*
         }
     };
-
-// No more tokens to munch
+    // No more tokens to munch
     (@arg ($arg:expr) $modes:tt) => { $arg };
-// Shorthand tokens influenced by the usage_string
+    // Shorthand tokens influenced by the usage_string
     (@arg ($arg:expr) $modes:tt --($long:expr) $($tail:tt)*) => {
         $crate::clap_app!{ @arg ($arg.long($long)) $modes $($tail)* }
     };
+    (@arg ($arg:expr) $modes:tt --$long:literal $($tail:tt)*) => {
+        $crate::clap_app!{ @arg ($arg.long(stringify!($long).trim_matches('"'))) $modes $($tail)* }
+    };
     (@arg ($arg:expr) $modes:tt --$long:ident $($tail:tt)*) => {
         $crate::clap_app!{ @arg ($arg.long(stringify!($long))) $modes $($tail)* }
     };
+    (@arg ($arg:expr) $modes:tt -($short:expr) $($tail:tt)*) => {
+        $crate::clap_app!{ @arg ($arg.short($short.chars().next().unwrap())) $modes $($tail)* }
+    };
+    (@arg ($arg:expr) $modes:tt -$short:literal $($tail:tt)*) => {
+        $crate::clap_app!{ @arg ($arg.short($short.to_string().chars().next().unwrap())) $modes $($tail)* }
```

Comment:
More user friendly message on `""`

```suggestion
        $crate::clap_app!{ @arg ($arg.short($short.to_string().chars().next().expect(r#""" is not allowed here"#))) $modes $($tail)* }
```